mcp/sensors/star_camera_receive.c

```diff
@@ -0,0 +1,347 @@
+        sc1_hm_Addr = channels_find_by_name("sc1_hm");
+        sc1_focus_position_Addr = channels_find_by_name("sc1_focus_position");
+        sc1_focus_inf_Addr = channels_find_by_name("sc1_focus_inf");
+        sc1_aperture_steps_Addr = channels_find_by_name("sc1_aperture_steps");
+        sc1_max_aperture_Addr = channels_find_by_name("sc1_max_aperture");
+        sc1_min_focus_pos_Addr = channels_find_by_name("sc1_min_focus_pos");
+        sc1_max_focus_pos_Addr = channels_find_by_name("sc1_max_focus_pos");
+        sc1_current_aperture_Addr = channels_find_by_name("sc1_current_aperture");
+        sc1_exposure_time_Addr = channels_find_by_name("sc1_exposure_time");
+        sc1_focus_mode_Addr = channels_find_by_name("sc1_focus_mode");
+        sc1_start_focus_pos_Addr = channels_find_by_name("sc1_start_focus_pos");
+        sc1_end_focus_pos_Addr = channels_find_by_name("sc1_end_focus_pos");
+        sc1_focus_step_Addr = channels_find_by_name("sc1_focus_step");
+        sc1_photos_per_focus_Addr = channels_find_by_name("sc1_photos_per_focus");
+    }
+    SET_SCALED_VALUE(sc1_spike_limit_Addr, scr.blobParams[0]);
+    SET_SCALED_VALUE(sc1_dyanmic_hp_Addr, scr.blobParams[1]);
+    SET_SCALED_VALUE(sc1_r_smooth_Addr, scr.blobParams[2]);
+    SET_SCALED_VALUE(sc1_high_pass_filter_Addr, scr.blobParams[3]);
+    SET_SCALED_VALUE(sc1_r_high_pass_filter_Addr, scr.blobParams[4]);
+    SET_SCALED_VALUE(sc1_search_border_Addr, scr.blobParams[5]);
+    SET_SCALED_VALUE(sc1_filter_return_Addr, scr.blobParams[6]);
+    SET_SCALED_VALUE(sc1_n_sigma_Addr, scr.blobParams[7]);
+    SET_SCALED_VALUE(sc1_unique_star_spacing_Addr, scr.blobParams[8]);
+    SET_SCALED_VALUE(sc1_make_static_hp_mask_Addr, scr.makeHP);
+    SET_SCALED_VALUE(sc1_use_static_hp_mask_Addr, scr.useHP);
+    SET_SCALED_VALUE(sc1_time_limit_Addr, scr.solveTimeLimit);
+    SET_SCALED_VALUE(sc1_logodds_Addr, scr.logOdds);
+    SET_SCALED_VALUE(sc1_latitude_Addr, scr.latitude);
+    SET_SCALED_VALUE(sc1_longitude_Addr, scr.longitude);
+    SET_SCALED_VALUE(sc1_hm_Addr, scr.heightWGS84);
+    SET_SCALED_VALUE(sc1_focus_position_Addr, scr.focusPos);
+    SET_SCALED_VALUE(sc1_focus_inf_Addr, scr.setFocusInf);
+    SET_SCALED_VALUE(sc1_aperture_steps_Addr, scr.apertureSteps);
+    SET_SCALED_VALUE(sc1_max_aperture_Addr, scr.maxAperture);
+    SET_SCALED_VALUE(sc1_min_focus_pos_Addr, scr.minFocusPos);
+    SET_SCALED_VALUE(sc1_max_focus_pos_Addr, scr.maxFocusPos);
+    SET_SCALED_VALUE(sc1_current_aperture_Addr, scr.aperture);
+    SET_SCALED_VALUE(sc1_exposure_time_Addr, scr.exposureTime);
+    SET_SCALED_VALUE(sc1_focus_mode_Addr, scr.focusMode);
+    SET_SCALED_VALUE(sc1_start_focus_pos_Addr, scr.startPos);
+    SET_SCALED_VALUE(sc1_end_focus_pos_Addr, scr.endPos);
+    SET_SCALED_VALUE(sc1_focus_step_Addr, scr.focusStep);
+    SET_SCALED_VALUE(sc1_photos_per_focus_Addr, scr.photosPerStep);
+}
+
+// function that runs in the receive thread to put data into the proper channels
```

Comment:
Good function comment, but use doxygen styling pls. And differentiate comments sc1 vs. sc2!

Reply:
do let me know if any comments in the next commit batch missed SC1/SC2 differentiation, I think I got them but its easy to miss one when writing 30